src/format.rs

```diff
@@ -62,39 +66,41 @@ pub fn handle_indentation(raw: &str) -> Option<String> {
 /// H4 -> H6
 /// H6 -> H6
 ///
-pub fn shift_headings(raw: &str, levels: usize) -> String {
-    let mut result = String::new();
-    for line in raw.split_inclusive('\n') {
-        if line.trim_start().starts_with('#') {
-            result.push_str(&handle_heading(line, levels));
-        } else {
-            result.push_str(line);
-        }
-    }
-    result
-}
+pub fn shift_headings(raw: &str, levels: u8) -> String {
+    let arena = Arena::new();
+
+    // Parse the document into an AST
+    let root = parse_document(&arena, raw, &ComrakOptions::default());
+    increase_heading_levels(root, levels);
+
+    let mut markdown_output = vec![];
+
+    // Change some of the default formatting options for better compatibility with nixos-render-docs (nrd).
+    let mut options: Options = ComrakOptions::default();
 
-// Dumb heading parser.
-pub fn handle_heading(line: &str, levels: usize) -> String {
-    let chars = line.chars();
+    // Each newline in the doc-comment sources create a visual line-break in the output,
+    // Making it more intuitive for those who expect a more "what you see is what you get" behavior.
+    // This doesnt conflict with the markdown spec, as it allows for hardbreaks. And allows to save vertical space in doc-comments.
+    options.render.hardbreaks = true;
 
-    // let mut leading_trivials: String = String::new();
-    let mut hashes = String::new();
-    let mut rest = String::new();
-    for char in chars {
-        match char {
-            '#' if rest.is_empty() => {
-                // only collect hashes if no other tokens
-                hashes.push(char)
+    // Disable automatic generation of header IDs. nrd will generate them.
+    options.extension.header_ids = None;
+
+    comrak::format_commonmark(root, &ComrakOptions::default(), &mut markdown_output).unwrap();
+    let markdown_output = String::from_utf8(markdown_output).unwrap();
+
+    return markdown_output;
```

Comment:
```suggestion
    String::from_utf8(markdown_output).unwrap()
```

Also some justification for the `unwrap`'s would be appreciated